inc/user.class.php

```diff
@@ -5558,6 +5558,47 @@ public function hasPasswordExpired() {
       return $expiration_time < time();
    }
 
+   public static function getFriendlyNameSearchCriteria(string $filter): array {
+      $table     = self::getTable();
+      $login     = DBmysql::quoteName("$table.name");
+      $firstname = DBmysql::quoteName("$table.firstname");
+      $lastname  = DBmysql::quoteName("$table.realname");
+
+      $filter = strtolower($filter);
+      $filter_no_spaces = str_replace(" ", "", $filter);
+
+      return [
+         'OR' => [
+            ['RAW' => ["LOWER($login)" => ['LIKE', "%$filter%"]]],
+            ['RAW' => ["LOWER(CONCAT($firstname, $lastname))" => ['LIKE', "%$filter_no_spaces%"]]],
+            ['RAW' => ["LOWER(CONCAT($lastname, $firstname))" => ['LIKE', "%$filter_no_spaces%"]]],
+         ]
+      ];
```

Comment:
It is common to have spaces in firstnames or lastnames, so you should not remove spaces here.

Reply:
I've removed spaces from the concat result instead.
The goal is to have a more forgiving search result so you can still match  "Adrien Clairembault" with "adrienclairembault".

After removing spaces from concat, it works fine for names with spaces:
![image](https://user-images.githubusercontent.com/42734840/88290891-b9896880-ccf7-11ea-96cb-75945c36e23f.png)